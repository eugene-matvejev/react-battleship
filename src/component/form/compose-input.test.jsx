import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import composeInput from './compose-input';

configure({ adapter: new Adapter() });

const ComposedInput = composeInput((props) => <input {...props} />);

describe('<ComposedInput/>', () => {
    const props = {
    };

    describe('render', () => {
        it('with default/required props', () => {
            const c = shallow(<ComposedInput {...props} />);

            expect(c).toMatchSnapshot();
        });

        describe('with optional props', () => {
            [
                ['errors', ['{{error}}']],
            ].forEach(([prop, v]) => {
                it(`[::${prop}] as "${v}"`, () => {
                    const c = shallow(<ComposedInput {...props} {...{ [prop]: v }} />);

                    expect(c).toMatchSnapshot();
                });
            });
        });
    });
});
