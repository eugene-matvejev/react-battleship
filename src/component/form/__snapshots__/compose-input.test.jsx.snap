// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComposedInput/> render with default/required props 1`] = `
<div
  className="form-input "
>
  <label
    className="form-input_label"
  >
    <Component />
  </label>
</div>
`;

exports[`<ComposedInput/> render with optional props [::errors] as "{{error}}" 1`] = `
<div
  className="form-input "
>
  <label
    className="form-input_label"
  >
    <Component />
    <ul
      className="form-input_errors"
    >
      <li
        className="form-input_error"
        data-cy="undefined-error-0"
        key="0"
      >
        {{error}}
      </li>
    </ul>
  </label>
</div>
`;
